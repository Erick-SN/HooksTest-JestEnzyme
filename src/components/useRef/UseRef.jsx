import React, { useState } from 'react';
import CustomHooks from '../useFetch/CustomHooks';
const UseRef = () => {
  const [show, setShow] = useState(false);
  return (
    <>
      <h1> useRef </h1>
      <hr />
      {show && <CustomHooks />}
      <button className='btn btn-primary mt-5' onClick={() => setShow(!show)}>
        Show / Hide
      </button>
    </>
  );
};

export default UseRef;
