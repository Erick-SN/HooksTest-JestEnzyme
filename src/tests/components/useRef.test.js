import React from 'react';
import { shallow } from 'enzyme';
import UseRef from '../../components/useRef/UseRef';
describe('Test at UseRef', () => {
  const wrapper = shallow(<UseRef />);
  test('should ', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('CustomHooks').exists()).toBe(false);
  });
  test('should  show CustomHooks', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('CustomHooks').exists()).toBe(true);
  });
});
