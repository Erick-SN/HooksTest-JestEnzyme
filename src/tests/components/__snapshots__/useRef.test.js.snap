// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at UseRef should  1`] = `
<Fragment>
  <h1>
     useRef 
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</Fragment>
`;
